src/App.tsx

```diff
@@ -1,24 +1,219 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo, UpdatedTodo } from './types/Todo';
+import {
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from './api/todos';
+import { Error } from './components/Error/Error';
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer/Footer';
+import { Todolist } from './components/Todolist/Todolist';
+import { ErrorMessage } from './types/ErrorMessage';
+import { TodoStatus } from './types/TodoStatus';
 
-const USER_ID = 0;
+const USER_ID = 10890;
 
 export const App: React.FC = () => {
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<TodoStatus>(TodoStatus.all);
+  const [errorMessage, setErrorMessage]
+  = useState<ErrorMessage>(ErrorMessage.noError);
+  const [isLoading, setIsLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [delUpdTodoIds, setDelUpdTodoIds] = useState<number[]>([]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setVisibleTodos)
+      .catch(() => {
+        setErrorMessage(ErrorMessage.loadError);
+      });
+  }, []);
+
+  const handleCloseError = () => {
+    setErrorMessage(ErrorMessage.noError);
+  };
+
+  const filteredTodos = useMemo(() => {
+    switch (filter) {
+      case TodoStatus.completed:
+        return visibleTodos.filter(todo => todo.completed);
+
+      case TodoStatus.active:
+        return visibleTodos.filter(todo => !todo.completed);
+
+      default:
+        return visibleTodos;
+    }
+  }, [filter, visibleTodos]);
+
+  const completedTodos = filteredTodos.filter(todo => todo.completed);
+  const activeTodos = filteredTodos.filter(todo => !todo.completed);
+
+  // const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+  //   e.preventDefault();
+
+  //   setIsLoading(true);
+  //   setTempTodo({
+  //     id: 0,
+  //     title: newTodoTitle,
+  //     completed: false,
+  //     userId: USER_ID,
+  //   });
+
+  //   if (!newTodoTitle.trim()) {
+  //     setErrorMessage(ErrorMessage.titleError);
+  //     setIsLoading(false);
+  //     setTempTodo(null);
+
+  //     return;
+  //   }
+
+  //   addTodos(USER_ID, {
+  //     title: newTodoTitle,
+  //     userId: USER_ID,
+  //     completed: false,
+  //   })
+  //     .then((result) => {
+  //       setVisibleTodos((prevTodos) => {
+  //         return [result, ...prevTodos];
+  //       });
```

Comment:
You don't want to see commented code. It is confusing af, as you never now what the code owner wanted to tell with it